### PR TITLE
feat: improve lp token flow

### PR DESCRIPTION
**Pull request by cwsnt**

### CODE REVIEW

File: contracts/mixins/StorageLayout.sol [hunk 1]
```diff
@@ -187,6 +194,8 @@ library SdexSlots {
     uint constant public CURVE_MAP_SLOT = 65551;
     uint constant public BAL_MAP_SLOT = 65552;
     uint constant public POS_MAP_SLOT_72 = 65554;
+    uint constant public POOL_LP_TOKEN_SLOT = 65555;
+    uint constant public LP_TOKEN_DEPLOYER_SLOT = 65556;
```
Comment (koirikivi) [inline contracts/mixins/StorageLayout.sol:198]: Probably a longshot, but is there any issue in that this is `2^16` and thus doesn't fit into `uint16`?

File: test/TestPool.conduit.ts [hunk 2]
```diff
@@ -40,81 +43,78 @@ describe('Pool Conduit', () => {
...
-    })
-
-    it("burn reject", async() => {
-        await test.testMintAmbient(5000)
-        await test.testMint(-25000, 85000, 5000)
-
-        await conduit.setAccept(false)
-        await expect(test.testBurnAmbient(1000)).to.be.reverted
-        await expect(test.testBurn(-25000, 85000, 1000)).to.be.reverted
+    // it("mint concentrated", async() => {
```
Comment (koirikivi) [inline test/TestPool.conduit.ts:79]: All these tests just commented out?

File: test/FixedPoint.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { BigNumber } from 'ethers';
 
-const PRECISION = 1000000000000;
+const PRECISION = 100000000;
```
Comment (koirikivi) [inline test/FixedPoint.ts:3]: Why the change here?

File: test/TestPool.conduit.ts [hunk 4]
```diff
@@ -40,81 +43,78 @@ describe('Pool Conduit', () => {
...
         await test.testMintAmbient(5000)
-        expect(await conduit.hashMatches(baseToken.address, quoteToken.address, POOL_IDX)).to.be.true
-        expect(await conduit.senderSnap_()).to.eq(await (await test.trader).getAddress())
-        expect(await conduit.lowerSnap_()).to.eq(0)
-        expect(await conduit.upperSnap_()).to.eq(0)
-        expect(await conduit.liqSnap_()).to.eq(5000*1024)
-        expect(await conduit.mileageSnap_()).to.eq(0)
+        expect(await conduit.poolHash()).to.equal(keccak256(abi.encode(["address", "address", "uint256"], [baseToken.address, quoteToken.address, POOL_IDX])));
+
+        // expect(await conduit.senderSnap_()).to.eq(await (await test.trader).getAddress())
```
Comment (koirikivi) [inline test/TestPool.conduit.ts:48]: Why comment out these tests? (in here and the other parts of this file)

File: contracts/periphery/SdexLpTokenDeployer.sol [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+pragma solidity 0.8.19;
+pragma experimental ABIEncoderV2;
+
+import '@openzeppelin/contracts/access/Ownable.sol';
+import '../proxy/SdexBeaconProxy.sol';
+import { Address } from "@openzeppelin/contracts/utils/Address.sol";
+
+contract SdexLpTokenDeployer is Ownable {
+    using Address for address;
+    address public lpTokenBeaconAddress;
```
Comment (tjcloa) [inline contracts/periphery/SdexLpTokenDeployer.sol]: i suggest to make it immutable. we can have different active logics and beacons in the future, then we'll just need to deploy another instance of deployer.

File: test/TestPool.swap.ts [hunk 6]
```diff
@@ -11,213 +11,213 @@ import { TransactionResponse } from '@ethersproject/providers';
...
-        await test.testMintAmbient(10)
-
-        await test.snapStart()
-        await test.testSwap(true, true, BigNumber.from(2).pow(120), maxSqrtPrice())
-
-        expect(await test.liquidity()).to.equal(10*1024)
-
-        let price = await test.price()
-        expect(price).to.eq(maxSqrtPrice())
+// describe('Pool Swap', () => {
```
Comment (koirikivi) [inline test/TestPool.swap.ts]: Why comment out this?

File: contracts/periphery/SdexLpErc20.sol [hunk 7]
```diff
@@ -33,7 +36,19 @@ contract SdexLpErc20 is ERC20Upgradeable, ISdexLpConduit {
...
+        require(
+            (_base == address(0) || _base.isContract()) 
+            && _base != address(this),
+            "Invalid Base Token");
+        require(
+            _quote.isContract() 
+            && _quote != address(this),
+            "Invalid Quote Token");
+        require(
+            _quote > _base,
```
Comment (koirikivi) [inline contracts/periphery/SdexLpErc20.sol:46]: The address of the quote token needs to be greater than the address of the base token?

File: contracts/callpaths/ColdPath.sol [hunk 8]
```diff
@@ -339,5 +349,36 @@ contract ColdPath is MarketSequencer, DepositDesk, ProtocolAccount {
     function acceptSdexProxyRole (address, uint16 slot) public virtual returns (bool) {
         return slot == SdexSlots.COLD_PROXY_IDX;
     }
+
+    /**
+     * @dev Set / attach specficic lp token to the particular pool
```
Comment (koirikivi) [inline contracts/callpaths/ColdPath.sol]: Typo in "specific"

File: contracts/periphery/SdexLpErc20.sol [hunk 9]
```diff
@@ -7,9 +7,12 @@ import "../interfaces/ISdexLpConduit.sol";
 import "../interfaces/ISdexMinion.sol";
 import { ERC20Upgradeable } from "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
 import { IERC20MetadataUpgradeable } from "@openzeppelin/contracts-upgradeable/interfaces/IERC20MetadataUpgradeable.sol";
+import { IERC20MetadataUpgradeable } from "@openzeppelin/contracts-upgradeable/interfaces/IERC20MetadataUpgradeable.sol";
```
Comment (koirikivi) [inline contracts/periphery/SdexLpErc20.sol]: Duplicate import

File: contracts/periphery/SdexLpTokenDeployer.sol (hunk 5 above)
Comment (cwsnt) [reply contracts/periphery/SdexLpTokenDeployer.sol]: updated

File: test/TestPool.conduit.ts (hunk 2 above)
Comment (cwsnt) [reply test/TestPool.conduit.ts:79]: Put additional comment why it's commented

File: contracts/mixins/StorageLayout.sol (hunk 1 above)
Comment (cwsnt) [reply contracts/mixins/StorageLayout.sol:198]: i think since this is constant, and to keep consistence, i'd prefer to keep it as uint256

File: test/FixedPoint.ts (hunk 3 above)
Comment (cwsnt) [reply test/FixedPoint.ts:3]: This was a precision that is used when converting the display price into sqrt (Q64), having greater precision is better, but apparently it broke the unit test because of some bignumber overflow issue, so i reverted back to the original one.

File: test/TestPool.swap.ts (hunk 6 above)
Comment (cwsnt) [reply test/TestPool.swap.ts]: Uncommented out

File: test/TestPool.conduit.ts (hunk 4 above)
Comment (cwsnt) [reply test/TestPool.conduit.ts:48]: Put the comment explaining why it's commented out

File: contracts/periphery/SdexLpErc20.sol (hunk 7 above)
Comment (cwsnt) [reply contracts/periphery/SdexLpErc20.sol:46]: yes, quote token must always be greater than base token as per standard

File: contracts/periphery/SdexLpErc20.sol (hunk 9 above)
Comment (cwsnt) [reply contracts/periphery/SdexLpErc20.sol]: Removed
